service/history/queues/scheduler.go

```diff
@@ -137,116 +132,30 @@ func NewNamespacePriorityScheduler(
 		)[key.Priority]
 	}
 	channelWeightUpdateCh := make(chan struct{}, 1)
-	channelQuotaRequestFn := func(key TaskChannelKey) quotas.Request {
-		namespaceName, err := namespaceRegistry.GetNamespaceName(namespace.ID(key.NamespaceID))
-		if err != nil {
-			namespaceName = namespace.EmptyName
-		}
-		return quotas.NewRequest("", taskSchedulerToken, namespaceName.String(), tasks.PriorityName[key.Priority], 0, "")
-	}
-	taskChannelMetricsTagsFn := func(key TaskChannelKey) []metrics.Tag {
-		namespaceName, _ := namespaceRegistry.GetNamespaceName(namespace.ID(key.NamespaceID))
-		return []metrics.Tag{
-			metrics.NamespaceTag(string(namespaceName)),
-			metrics.TaskPriorityTag(key.Priority.String()),
-		}
-	}
 	fifoSchedulerOptions := &tasks.FIFOSchedulerOptions{
 		QueueSize:   prioritySchedulerProcessorQueueSize,
 		WorkerCount: options.WorkerCount,
 	}
 
-	return &schedulerImpl{
-		Scheduler: tasks.NewInterleavedWeightedRoundRobinScheduler(
-			tasks.InterleavedWeightedRoundRobinSchedulerOptions[Executable, TaskChannelKey]{
-				TaskChannelKeyFn:            taskChannelKeyFn,
-				ChannelWeightFn:             channelWeightFn,
-				ChannelWeightUpdateCh:       channelWeightUpdateCh,
-				ChannelQuotaRequestFn:       channelQuotaRequestFn,
-				TaskChannelMetricTagsFn:     taskChannelMetricsTagsFn,
-				EnableRateLimiter:           options.EnableRateLimiter,
-				EnableRateLimiterShadowMode: options.EnableRateLimiterShadowMode,
-				DispatchThrottleDuration:    options.DispatchThrottleDuration,
-			},
-			tasks.Scheduler[Executable](tasks.NewFIFOScheduler[Executable](
-				fifoSchedulerOptions,
-				logger,
-			)),
-			rateLimiter,
-			timeSource,
+	scheduler = tasks.NewInterleavedWeightedRoundRobinScheduler(
+		tasks.InterleavedWeightedRoundRobinSchedulerOptions[Executable, TaskChannelKey]{
+			TaskChannelKeyFn:      taskChannelKeyFn,
+			ChannelWeightFn:       channelWeightFn,
+			ChannelWeightUpdateCh: channelWeightUpdateCh,
+		},
+		tasks.Scheduler[Executable](tasks.NewFIFOScheduler[Executable](
+			fifoSchedulerOptions,
 			logger,
-			metricsHandler,
-		),
-		namespaceRegistry:     namespaceRegistry,
-		taskChannelKeyFn:      taskChannelKeyFn,
-		channelWeightFn:       channelWeightFn,
-		channelWeightUpdateCh: channelWeightUpdateCh,
-	}
-}
-
-// NewPriorityScheduler ignores namespace when scheduleing tasks.
-// currently only used for shard level task scheduler
-func NewPriorityScheduler(
```

Comment:
I removed this scheduler because the per namespace rate limiter is now used for low priority tasks/queues (archival) as well, so doesn't make sense to use a scheduler that ignores namespaces.